exporter/exporterhelper/queue_sender.go

```diff
@@ -135,18 +141,17 @@ func (qs *queueSender) onTemporaryFailure(ctx context.Context, req Request, err
 
 // Start is invoked during service startup.
 func (qs *queueSender) Start(ctx context.Context, host component.Host) error {
-	err := qs.queue.Start(ctx, host, internal.QueueSettings{
-		DataType: qs.signal,
-		Callback: func(qr internal.QueueRequest) {
-			_ = qs.nextSender.send(qr.Context, qr.Request.(Request))
-			// TODO: Update OnProcessingFinished to accept error and remove the retry->queue sender callback.
-			qr.OnProcessingFinished()
-		},
-	})
+	err := qs.queue.Start(ctx, host)
 	if err != nil {
 		return err
 	}
 
+	qs.consumers = internal.NewConsumers(qs.queue, qs.numConsumers, func(item internal.QueueRequest) {
+		// TODO: Update item.OnProcessingFinished to accept error and remove the retry->queue sender callback.
```

Comment:
Where do you call `item.OnProcessingFinished` now? Would be good if Consumers take care of data and you just pass a func that accepts context and request.

Reply:
Yes, it can be nicely done with the generics now. Applied 👍